_posts/2019-02-28-Solving-Switch-Riddle-using-RL.md

```diff
@@ -0,0 +1,117 @@
+﻿---
+layout: post
+title: "Solving Switch Riddle using RL"
+author_github: Madhuparna04
+date: 2019-02-28
+categories: Reinforcement Learning
+author: Madhuparna Bhowmik
+---
+
+In this article, we will use a Reinforcement Learning based method to solve the Switch Riddle problem. So, let's start with defining the riddle.
+
+
+## Switch Riddle
+
+There are n prisoners in prison and a warden. The Warden decides to free the prisoners if they can solve the following problem. So, every day the Warden will select one of the prisoners and send him to an interrogation room which consists of a light bulb with a switch. If the prisoner in the room can tell that all other prisoners including him have been to the room at least once then the Warden will free all of them otherwise kill all of them.
+Except for the prisoner in the room, other prisoners are unaware of the fact that who got selected on that particular day to go to the interrogation room.
+
+![Switch Riddle](/assets/SwitchRiddle.png "Switch Riddle")
+
+
+[Image Source](https://github.com/iassael/learning-to-communicate)
+
+Now, the prisoner in the interrogation room can switch on or off the bulb to send some indication to the next prisoner. He can also tell the warden that everyone has been to the room at least once or decide not to say anything. If his claim is correct, then all are set free otherwise no.
+
+## Problem Formulation
+
+So we see that the switch riddle problem consists of many prisoners trying to establish a single task. Therefore, the puzzle fits in a Multi-Agent setup where the agents are collaborating to complete the task.

```

Comment:
We can observe that switch riddle problem comprises of multiple agents working towards completing a single task.